modules/launchd/default.nix

```
@@ -209,5 +209,28 @@ in {
           setupLaunchAgents
         '';
     })
+
+    (mkIf isDarwin (let
+      setenv-script = pkgs.writeShellScript "launchctl-setenv"
+        (concatStringsSep "\n" (mapAttrsToList
+          (name: val: "/bin/launchctl setenv ${name} ${toString val}")
+          config.home.sessionVariables));
+      launchctl-setenv = pkgs.stdenv.mkDerivation {
+        name = "setup-script";
+        buildCommand = ''
+          mkdir -p $out/bin
+          cp ${setenv-script} $out/bin/launchctl-setenv
+        '';
+      };
```

Should be replaceable by

```suggestion
      launchctl-setenv = pkgs.writeShellScriptBin "launchctl-setenv"
        (concatStringsSep "\n" (mapAttrsToList
          (name: val: "/bin/launchctl setenv ${name} ${toString val}")
          config.home.sessionVariables));
      };
```